packages/main/src/Menu.ts

```diff
@@ -266,22 +273,73 @@ class Menu extends UI5Element {
 		return this.shadowRoot!.querySelector<ResponsivePopover>("[ui5-responsive-popover]")!;
 	}
 
+	get _list() {
+		return this.shadowRoot!.querySelector<List>("[ui5-list]")!;
+	}
+
+	get _menuItemGroups() {
+		return this.items.filter((item) : item is MenuItemGroup => item.isGroup);
+	}
+
+	// Return only the menu items, excluding separators and items that belong to groups
 	get _menuItems() {
-		return this.items.filter((item): item is MenuItem => !item.isSeparator);
+		return this.items.filter((item) : item is MenuItem => !item.isSeparator && !item.isGroup);
+	}
+
+	get _menuItemsAll() {
+		const items: MenuItem[] = [];
```

Comment:
Shouldn't the type be IMenuItem[], as MenuItemGroup does not implement MenuItem but it's assigned

Reply:
**MenuItemGroup** implements **IMenuItem** - this interface is needed to be implemented because **Menu**, **MenuItem** and **MenuItemGroup** `items` slot accepts **MenuItem**, **MenuSeparator** and **MenuItemGroup** components (_items!: Array<IMenuItem>;_). But `_menuItemsAll` getter returns all **MenuItem** components only by skipping **MenuSeparator** components and including **MenuItem** components slotted in the **MenuItemGroup** `items` slot. That's why the type should be **MenuItem**.

Reply:
We checked it in discussion, I missaw an assignment